src/ORM/Table.php

```diff
@@ -279,7 +279,12 @@ public function __construct(array $config = [])
             }
         }
         $this->_eventManager = $eventManager ?: new EventManager();
-        $this->_behaviors = $behaviors ?: new BehaviorRegistry($this);
+        if ($behaviors) {
+            $behaviors->setTable($this);
+        } else {
+            $behaviors = new BehaviorRegistry($this);
```

Comment:
You can avoid the if else entirely if you always use the setter and you restore the line you removed here (minus the constructor argument)